### ISSUE TITLE
Data asset names are not URL encoded for Lineage API endpoint requests

### ISSUE THREAD

**Issue by stevenspadotto** [open-metadata/OpenMetadata#8735]
**Affected module**

API and UI

**Describe the bug**

Lineage produces a 404 error when expanding a node that has a name which needs to be URL encoded (ex: a `fullyQualifiedName` with forward slashes in it). 

The issue is that data asset names are not URL encoded before passing them to the name parameter in the lineage API endpoint URL.

This API request returns an errror:

```
/api/v1/lineage/table/name/Example.Element.production./var/example/data/intermediate/element/?upstreamDepth=3&downstreamDepth=3
```

URL encoding the name parameter causes the API request to succeed:

```
/api/v1/lineage/table/name/Example.Element.production.%2Fvar%2Fexample%2Fdata%2Fintermediate%2Felement%2F?upstreamDepth=3&downstreamDepth=3
```

The same applies for the `/api/v1/lineage/pipeline/name/` endpoint.

**To Reproduce**

Go to Lineage and expand a node that has a `fullyQualifiedName` with characters that need to be URL encoded (ex: forward slashes)

**Expected behavior**

No 404 error returned by Lineage when expanding a node that has characters that need to be URL encoded (ex: forward slashes)

**Version:**
 - OS: OS X
 - Python version:
 - OpenMetadata version: 0.12.2
 - OpenMetadata Ingestion package version: 

**Additional context**

None

**Comment by pmbrull** [comment]
Thanks for reaching out! Let us take a look 🙏 

cc @harshach @chirag-madlani

**Comment by Ashish8689** [comment]
Hii @stevenspadotto,  we are already encoding it, please find below the example from Sandbox. Let me know if I am missing something.

`https://sandbox.open-metadata.org/table/demo_datalake.default.awsdatalake-testing.%22data%2Fsales%2Fsales.csv%22/lineage`

<img width="1184" alt="image" src="https://user-images.githubusercontent.com/66266464/202414239-bc944421-37a4-4e9f-be2e-39919f87e465.png">

**Comment by stevenspadotto** [comment]
@Ashish8689 the GET request that is being used when I try to expand a node in the lineage graph by clicking on the `<` symbol is the following. Here is what I see as the Request URL. The forward slashes are not URL encoded:

```
GET https://.../api/v1/lineage/pipeline/name/Provider./data/front/dimensions/some_service_dimension/?upstreamDepth=3&downstreamDepth=3
```

**Comment by harshach** [comment]
Thanks @stevenspadotto for reporting the issues. Its now fixed in 0.13.0 release
